File: canvas_parser/weekly_iteration/paths.py
"""Default paths for weekly iteration cache and reports."""

from __future__ import annotations

from pathlib import Path

CACHE_DIR_NAME = '.cache/weekly_iteration'


def cache_dir(root: Path) -> Path:
    return root / CACHE_DIR_NAME


def default_snapshot_path(root: Path) -> Path:
    return cache_dir(root) / 'snapshots_enriched.json'


def fixture_snapshot_path(root: Path) -> Path:
    return root / 'fixtures' / 'weekly_iteration' / 'snapshots_gt.json'


def default_graph_cache_path(root: Path) -> Path:
    return cache_dir(root) / 'graph_eval.json'


def default_report_path(root: Path) -> Path:
    return cache_dir(root) / 'report.json'


def holdout_ground_truth_dir(root: Path) -> Path:
    return root / 'ground-truth' / 'holdout'


def holdout_snapshot_path(root: Path) -> Path:
    return cache_dir(root) / 'snapshots_holdout.json'


def holdout_fixture_snapshot_path(root: Path) -> Path:
    return root / 'fixtures' / 'weekly_iteration' / 'snapshots_holdout.json'


def holdout_report_path(root: Path) -> Path:
    return cache_dir(root) / 'report_holdout.json'


def harvard_ground_truth_dir(root: Path) -> Path:
    return root / 'ground-truth' / 'harvard'


def harvard_snapshot_path(root: Path) -> Path:
    return cache_dir(root) / 'snapshots_harvard.json'


def harvard_fixture_snapshot_path(root: Path) -> Path:
    return root / 'fixtures' / 'weekly_iteration' / 'snapshots_harvard.json'


def harvard_report_path(root: Path) -> Path:
    return cache_dir(root) / 'report_harvard.json'

File: task_optimizer_iteration/paths.js
const path = require('path')

const CACHE_DIR_NAME = '.cache/task_optimizer'

function repoRoot(startPath = __dirname) {
  return path.resolve(startPath, '..')
}

function cacheDir(root = repoRoot()) {
  return path.join(root, CACHE_DIR_NAME)
}

function fixturePath(root = repoRoot()) {
  return path.join(root, 'fixtures', 'task_optimizer', 'scenarios_gt.json')
}

function defaultReportPath(root = repoRoot()) {
  return path.join(cacheDir(root), 'report.json')
}

function defaultTasksExportPath(root = repoRoot()) {
  return path.join(cacheDir(root), 'tasks_export.json')
}

function holdoutFixturePath(root = repoRoot()) {
  return path.join(root, 'fixtures', 'task_optimizer', 'scenarios_holdout.json')
}

function holdoutReportPath(root = repoRoot()) {
  return path.join(cacheDir(root), 'report_holdout.json')
}

module.exports = {
  CACHE_DIR_NAME,
  repoRoot,
  cacheDir,
  fixturePath,
  holdoutFixturePath,
  defaultReportPath,
  holdoutReportPath,
  defaultTasksExportPath
}
